src/__tests__/reducer.plugin.spec.js

```diff
@@ -115,6 +118,53 @@ const describePlugin = (vanillaReducer, expect, { fromJS, deleteIn }) => () => {
         }
       })
   })
+  
+  it('should be provided the state from before the vanillaReducer', () => {
+    const state1 = fromJS({
+      foo: {
+        values: {
+          cat: 'beta',
+          lastCat: 'alpha' 
+        },
+        fields: {
+          cat: { touched: false },
+          lastCat: { touched: false }
+        }
+      }
+    })
+
+    // this plugin will change the value we are after so we can confirm we get the real starting state
+    const intermediatePlugin = (state) => setIn(state, 'values.cat', 'zed')
+    
+    const plugin = (state, action, startingState) => {
```

Comment:
The name `startingState` was chosen because it made sense to me at the time but I expect core maintainers to have a better name for this argument.